WatsonWebserver/Parameters/RouteManager.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace WatsonWebserver
+{
+    public class RouteManager
```

Comment:
Hi @zsolt777 first of all thank you for submitting this PR.  I love the approach.  What I'd request be changed is the naming.  Watson has dynamic routes, static routes, content routes, and parameter routes (in addition to default, etc).  Calling the new model as simply 'routes' could be confusing.  Could you think of a more specific name we could use?

Reply:
Hi @jchristn!  :) Please suggest a name for it. I was thinking on that but.... You can see the result. The ParameterRoute and the DynamicRoute is "reserved". I have no idea how to name it. :)

Reply:
I believe ASP.NET WebAPI calls these attribute routes.  How would you feel about that?  https://docs.microsoft.com/en-us/aspnet/web-api/overview/web-api-routing-and-actions/attribute-routing-in-web-api-2

Reply:
My personal preference is to use GET and POST only. It seems cool to use DELETE and PUT methods also, but once when I had to use a 3rd party REST API, I didn't like that. 
Everything looked great at the first glance, but the result was a nightmare (for me of course). The same url (eg .../1/document) with POST meant "Add new", with PUT meant "Update" and with GET meant "Get", and DELETE of course delete. 
I prefer this way: .../document/AddNew, .../document/Update, .../document/Delete with parameters of course.
For my eyes this way is much more easier to read and is cleaner, and I don't have to check the type of the http method. The GET is used when I send simple data types, and the POST is used for complex data structures.
What to say. I'm sure a lot of people prefers this way and others that way...